packages/analytics/analytics-chart/src/components/chart-plugins/DragSelectPlugin.ts

```diff
@@ -17,12 +17,14 @@ const dispatchEvent = (eventName: string, chart: Chart, pluginInstance: DragSele
   const xStartValue = chart.scales.x.getValueForPixel(pluginInstance.startX)
   const xEndValue = chart.scales.x.getValueForPixel(pluginInstance.endX)
 
-  chart.canvas.dispatchEvent(new CustomEvent<DragSelectEventDetail>(eventName, {
-    detail: {
-      xStart: xStartValue,
-      xEnd: xEndValue,
-    },
-  }))
+  if (xStartValue && xEndValue) {
+    chart.canvas.dispatchEvent(new CustomEvent<DragSelectEventDetail>(eventName, {
+      detail: {
+        xStart: xStartValue < xEndValue ? xStartValue : xEndValue,
+        xEnd: xEndValue > xStartValue ? xEndValue : xStartValue,
```

Comment:
nit: Why not just use `Math.min` and `Math.max` here?

Reply:
Good idea! 😅